### PR TITLE
feat: adding Scala generator

### PR DESCRIPTION
**Pull request by artur-ciocanu**
**Description**
- Adding Scala generator

**Related issue(s)**
Resolves https://github.com/asyncapi/modelina/issues/1664

### CODE REVIEW

**Comment by artur-ciocanu** [comment]
The PR is WIP ...

**Comment by jonaslagoni** [comment]
Let me know if you need anything from me @artur-ciocanu 🤙 Looking forward to having this!

**Comment by jonaslagoni** [comment]
@artur-ciocanu you can ignore sonar duplication warning

File: package.json
Comment (jonaslagoni) [inline package.json:131]:
```suggestion
```

**Review by jonaslagoni — CHANGES_REQUESTED** [review]
Only thing left to do as I see is adding simple generate example: https://github.com/asyncapi/modelina/tree/master/examples/generate-java-models

There is a template you can copy and change 🙂

File: package.json
Comment (artur-ciocanu) [reply package.json]: Good catch 😄

**Comment by artur-ciocanu** [comment]
@jonaslagoni it seems that I have covered everything that is required for Scala code gen. Once this PR is out, I'll add some JSON serialization preset and Jakarta validation preset.

Let me know your thought and if there is anything else required.

Thank you!

**Review by jonaslagoni — APPROVED** [review]
Awesome @artur-ciocanu, thanks for taking the time to provide it 🤙 Got a couple of small things that needs to be added, but Ill create issues after, not required to merge 😄

Also, do you want to become the codeowner of the scala generator, or are you only looking to donate it? ✌️

**Comment by jonaslagoni** [comment]
@all-contributors please add @artur-ciocanu for code, example, docs, test

**Comment by artur-ciocanu** [comment]
> Awesome @artur-ciocanu, thanks for taking the time to provide it 🤙 Got a couple of small things that needs to be added, but Ill create issues after, not required to merge 😄
> 
> Also, do you want to become the codeowner of the scala generator, or are you only looking to donate it? ✌️

@jonaslagoni I can be a code owner, I would like to get rid of my "internal" fork and just maintain the official Modelina version.

Thanks for merging it!

**Comment by jonaslagoni** [comment]
> @jonaslagoni I can be a code owner, I would like to get rid of my \"internal\" fork and just maintain the official Modelina version.

Awesome, thanks for doing it! Added a PR to add you 🤙 https://github.com/asyncapi/modelina/pull/1686

**Pull request merged**
